esphome/const.py

```diff
@@ -337,6 +337,7 @@
 CONF_IF = "if"
 CONF_IGNORE_EFUSE_MAC_CRC = "ignore_efuse_mac_crc"
 CONF_IGNORE_OUT_OF_RANGE = "ignore_out_of_range"
+CONF_IGNORE_PIN_VALIDATION_WARNING = "ignore_pin_validation_warning"
```

Comment:
Should this be error instead of warning?
```suggestion
CONF_IGNORE_PIN_VALIDATION_ERROR = "ignore_pin_validation_error"
```